pkg/cmd/adm/install_operator_test.go

```diff
@@ -0,0 +1,264 @@
+			// given
+			fakeClient := test.NewFakeClient(t, &installPlan)
+			fakeClientWithReadyCatalogSource(fakeClient)
+			term := NewFakeTerminalWithResponse("Y")
+			ctx := clicontext.NewTerminalContext(term)
+
+			// when
+			err := installOperator(ctx, installArgs{
+				kubeConfig: kubeconfig,
+				namespace:  namespace,
+			},
+				operator,
+				timeout,
+				commonclient.NewApplyClient(fakeClient),
+			)
+
+			// then
+			require.NoError(t, err)
+			ns := &corev1.Namespace{}
+			require.NoError(t, fakeClient.Get(context.TODO(), types.NamespacedName{Name: namespace}, ns)) // check that the namespace was created as well
+			AssertCatalogSourceExists(t, fakeClient, types.NamespacedName{Name: fmt.Sprintf("%s-operator", operator), Namespace: namespace})
+			AssertCatalogSourceHasSpec(t, fakeClient, types.NamespacedName{Name: fmt.Sprintf("%s-operator", operator), Namespace: namespace},
+				olmv1alpha1.CatalogSourceSpec{
+					SourceType:  olmv1alpha1.SourceTypeGrpc,
+					Image:       fmt.Sprintf("quay.io/codeready-toolchain/%s-operator-index:latest", operator),
+					DisplayName: fmt.Sprintf("KubeSaw %s Operator", operator),
+					Publisher:   "Red Hat",
+					UpdateStrategy: &olmv1alpha1.UpdateStrategy{
+						RegistryPoll: &olmv1alpha1.RegistryPoll{
+							Interval: &metav1.Duration{
+								Duration: 5 * time.Minute,
+							},
+						},
+					},
+				},
+			)
+			AssertOperatorGroupExists(t, fakeClient, types.NamespacedName{Name: fmt.Sprintf("%s-operator", operator), Namespace: namespace})
+			AssertSubscriptionExists(t, fakeClient, types.NamespacedName{Name: fmt.Sprintf("%s-operator", operator), Namespace: namespace})
+			assert.Contains(t, term.Output(), fmt.Sprintf("The %s operator has been successfully installed in the %s namespace", operator, namespace))
+		})
+
+		t.Run("install "+operator+" operator fails if CatalogSource is not ready", func(t *testing.T) {
+			// given
+			fakeClient := test.NewFakeClient(t)
+			term := NewFakeTerminalWithResponse("Y")
+			ctx := clicontext.NewTerminalContext(term)
+
+			// when
+			err := installOperator(ctx, installArgs{
+				kubeConfig: kubeconfig,
+				namespace:  namespace,
+			},
+				operator,
+				timeout,
+				commonclient.NewApplyClient(fakeClient),
+			)
+
+			// then
+			require.EqualError(t, err, fmt.Sprintf("failed waiting for catalog source to be ready.\n CatalogSrouce found: {\"kind\":\"CatalogSource\",\"apiVersion\":\"operators.coreos.com/v1alpha1\",\"metadata\":{\"name\":\"%[1]s-operator\",\"namespace\":\"toolchain-%[1]s-operator\",\"resourceVersion\":\"1\",\"generation\":1,\"creationTimestamp\":null},\"spec\":{\"sourceType\":\"grpc\",\"image\":\"quay.io/codeready-toolchain/%[1]s-operator-index:latest\",\"updateStrategy\":{\"registryPoll\":{\"interval\":\"5m0s\"}},\"displayName\":\"KubeSaw %[1]s Operator\",\"publisher\":\"Red Hat\",\"icon\":{\"base64data\":\"\",\"mediatype\":\"\"}},\"status\":{}} \n\t", operator))
```

Comment:
you can simplify this just to
```suggestion
			require.ErrorContains(t, err, "failed waiting for catalog source to be ready.")
```
and the other ones as well. I don't think that we need to verify the whole dump of the resources

Reply:
good point, done in https://github.com/kubesaw/ksctl/pull/48/commits/fe53e0a06998ac4d724c34472665f79bd42d6b44